nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -214,16 +237,154 @@ def reset_tools(self):
         else:
             self.data_collector.reset()
         if self.metrics_calculator is None:
-            self.metrics_calculator = NormMetricsCalculator()
+            self.metrics_calculator = NormMetricsCalculator(block_sparse_size=self.block_sparse_size)
         if self.sparsity_allocator is None:
             if self.mode == "normal":
-                self.sparsity_allocator = NormalSparsityAllocator(self)
+                self.sparsity_allocator = NormalSparsityAllocator(self, block_sparse_size=self.block_sparse_size)
             elif self.mode == "balance":
                 assert self.balance_gran is not None, 'balance_gran should be passed as param in balance mode'
-                self.sparsity_allocator = BankSparsityAllocator(self, self.balance_gran)
+                self.sparsity_allocator = BankSparsityAllocator(self, self.balance_gran, block_sparse_size=self.block_sparse_size)
             else:
                 raise NotImplementedError('Only support mode `normal` and `balance`')
 
+
+class LevelPruner(_LevelPruner):
+    r"""
+    This is a basic pruner pruning the model in the granularity of single value. In some papers called it magnitude pruning or fine-grained pruning.
```

Comment:
In some papers, it is also called magnitude ...

Reply:
fixed